sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/grouping.scala

```diff
@@ -49,11 +50,11 @@ trait GroupingSet extends Expression with CodegenFallback {
       > SELECT name, age, count(*) FROM VALUES (2, 'Alice'), (5, 'Bob') people(age, name) GROUP BY _FUNC_(name, age);
         Bob	5	1
         Alice	2	1
-        NULL	NULL	2
-        NULL	5	1
-        Bob	NULL	1
         Alice	NULL	1
         NULL	2	1
+        NULL	NULL	2
+        Bob	NULL	1
+        NULL	5	1
   """,
```

Comment:
This pr can affect parition ids (different hash values for int/long), so it changes the output order.